sentry-uitest/sentry-uitest-android-benchmark/src/androidTest/java/io/sentry/uitest/android/benchmark/SentryBenchmarkTest.kt

```diff
@@ -0,0 +1,135 @@
+        // On slow devices, a difference of 1 fps on an average of 20 fps would account for 5% decrease.
+        // On even slower devices the difference would be even higher. Let's skip for now, as it's not important anyway.
+//        assertTrue { comparisonResult.fpsDecrease >= -2 }
+//        assertTrue { comparisonResult.fpsDecrease < 2 }
+        assertTrue { comparisonResult.droppedFramesIncrease >= -1 }
+        assertTrue { comparisonResult.droppedFramesIncrease < 1 }
+    }
+
+    @Test
+    fun benchmarkProfiledTransaction() {
+
+        runner.runOnMainSync {
+            SentryAndroid.init(context) { options: SentryOptions ->
+                options.dsn = "https://key@uri/1234567"
+                options.isEnableAutoSessionTracking = false
+                options.tracesSampleRate = 1.0
+                options.isTraceSampling = true
+                options.isProfilingEnabled = true
+            }
+        }
+
+        // We compare the same operation with and without profiled transaction.
+        // We expect the profiled transaction operation to be slower, but not slower than 5%.
+        val benchmarkOperationNoTransaction = BenchmarkOperation(runner, getOperation(runner) { null })
+        val benchmarkOperationProfiled = BenchmarkOperation(
+            runner,
+            getOperation(runner) {
+                Sentry.startTransaction("Benchmark", "ProfiledTransaction")
+            }
+        )
+        val comparisonResult = BenchmarkOperation.compare(
+            benchmarkOperationNoTransaction,
+            "NoTransaction",
+            benchmarkOperationProfiled,
+            "ProfiledTransaction"
+        )
+
+        assertTrue { comparisonResult.durationIncrease >= 0 }
+        assertTrue { comparisonResult.durationIncrease < 5.0 }
+        assertTrue { comparisonResult.cpuTimeIncrease >= 0 }
+        assertTrue { comparisonResult.cpuTimeIncrease < 5.0 }
+        assertTrue { comparisonResult.fpsDecrease >= 0 }
+        assertTrue { comparisonResult.fpsDecrease < 5.0 }
+        assertTrue { comparisonResult.droppedFramesIncrease >= 0 }
+        assertTrue { comparisonResult.droppedFramesIncrease < 5.0 }
+    }
+
+    /**
+     * Operation that will be compared: it launches [BenchmarkActivity], swipe the list and closes it.
+     * The [transactionBuilder] is used to create the transaction before the swipes.
+     */
+    private fun getOperation(runner: AndroidJUnitRunner, transactionBuilder: () -> ITransaction?): () -> Unit = {
+        var transaction: ITransaction? = null
+        // Launch the sentry-uitest-android-benchmark activity
+        val benchmarkScenario = launchActivity<BenchmarkActivity>()
+        // Starts a transaction (it can be null, but we still runOnMainSync to make operations as similar as possible)
+        runner.runOnMainSync {
+            transaction = transactionBuilder()
+        }
+        // Just swipe the list some times: this is the benchmarked operation
+        repeat(2) {
```

Comment:
maybe extract this into a private method (or can be even a local function), like:
```kotlin
        val swipe = fun(times: Int) {
            repeat(times) {
		onView
     		onIdle
	    }
        }
```